### PR TITLE
Make xhistogram dependency optional

### LINKED ISSUE #82: Make xhistogram optional
Given the wide range of utilities we want to add to pymc-experimental it would be great if any dependencies that are not bundled with PyMC were optional. If I don't intend to use the histogram functionality, I would rather not have to install xhistogram.

CC @ferrine

### PR DESCRIPTION
**Pull request by ricardoV94**
Closes #82

### CODE REVIEW

**Comment by ferrine** [comment]
let's add to a setup.py an extention

```python
extras=dict(histogram=[...], complete=[...])
```

So one can install a specific set of dependencies

**Comment by ricardoV94** [comment]
> let's add to a setup.py an extention
> 
> ```python
> extras=dict(histogram=[...], complete=[...])
> ```
> 
> So one can install a specific set of dependencies

Do you want to push that into this PR? I have no idea how that's done but sounds quite good

File: setup.py [hunk 1]
```diff
@@ -81,5 +81,9 @@ def get_version():
         classifiers=classifiers,
         python_requires=">=3.8",
         install_requires=install_reqs,
-        extras_requires=dict(dask=["dask[all]"]),
+        extras_requires=dict(
+            dask=["dask[all]"], 
+            histogram=["xhistogram"], 
+            complete=["dask[all]", "xhistogram"]
```
Comment (ricardoV94) [inline setup.py:87]: Should it be? Or does `complete` have a special meaning

```suggestion
            all=["dask[all]", "xhistogram"]
```

**Comment by ferrine** [comment]
looks good to me

**Review by ferrine — APPROVED** [review]

**Comment by ricardoV94** [comment]
Thanks @ferrine!

**Comment by ferrine** [comment]
I noticed dask requirement on its own does not make sense so I've incremented the dependency list

**Comment by ferrine** [comment]
+ there is warning
```
WARNING: dask 2022.11.0 does not provide the extra 'all'
```

**Review by ferrine — APPROVED** [review]

**Pull request merged**